tracer/src/Datadog.Tracer.Native/cor_profiler.cpp

```diff
@@ -684,11 +684,22 @@ HRESULT CorProfiler::TryRejitModule(ModuleID module_id)
     {
         if (module_info.assembly.name.rfind(skip_assembly_pattern, 0) == 0)
         {
+            // The assembly matches the "skip" prefix, but check if it's specifically included
+            for (auto&& include_assembly : include_assemblies)
+            {
+                if (module_info.assembly.name == include_assembly)
+                {
+                    Logger::Debug("ModuleLoadFinished matched module by pattern: ", module_id, " ", module_info.assembly.name,
+                        "but assembly is explicitly included");
+                    goto inject;
```

Comment:
😲 😲 😲 😲 😲

Reply:
You can leave it like this or or or use a `break;` to prevent my eyes from bleeding.

Reply:
Can't use a `break` because we're two for-loops in at this point 😛

Reply:
There's ways of removing the `break` if it's too offensive 😉 

We could check if the assembly is in the `include_assembly` list _first_. Only check the `skip_assembly_prefixes` if it's not in that list.

Given that _most_ assemblies _won't_ be on the list, the approach approach I used should have less overhead I think, by avoiding the extra loop for _every_ assembly? It's a tiny difference though 😉 Happy to change it if you think that's preferable! 🙂

Reply:
🤦🏻 true.

Reply:
or add the typical bool "found" and `if (found) {break}` once again... 🙄🙈